libtock/screen.c

```diff
@@ -0,0 +1,246 @@
+#include "screen.h"
+#include <stdlib.h>
+#include <tock.h>
+
+typedef struct {
+  int error;
+  int data1;
+  int data2;
+  bool done;
+} ScreenReturn;
+
+static void screen_callback(int error,
+                            int data1,
+                            int data2,
+                            void* ud) {
+  ScreenReturn *fbr = (ScreenReturn*)ud;
+  fbr->error = error;
+  fbr->data1 = data1;
+  fbr->data2 = data2;
+  fbr->done  = true;
+}
+
+static uint8_t *buffer   = NULL;
+static size_t buffer_len = 0;
+
+static int screen_subscribe (subscribe_cb cb, void *userdata) {
+  return subscribe(DRIVER_NUM_SCREEN, 0, cb, userdata);
+}
+
+static int screen_command (int command_num, int data1, int data2) {
+  return command(DRIVER_NUM_SCREEN, command_num, data1, data2);
+}
+
+static int screen_allow (void* ptr, size_t size) {
+  return allow(DRIVER_NUM_SCREEN, 0, ptr, size);
+}
+
+int screen_get_supported_resolutions (void) {
+  ScreenReturn fbr;
+  fbr.data1 = 0;
+  fbr.done  = false;
+  screen_subscribe (screen_callback, &fbr);
+  fbr.error = screen_command (11, 0, 0);
+  if (fbr.error == TOCK_SUCCESS) yield_for (&fbr.done);
+  return fbr.data1;
```

Comment:
Should this return `fbr.error` in the case that the original command error'd? (Here, and later throughout this file)

Reply:
This is correct, as the function returns the number of available resolutions.